src/uwtools/config/atparse_to_jinja2.py

```diff
@@ -3,14 +3,14 @@
 """
 
 import re
-from typing import IO, Any, Generator, Optional
+from typing import IO, Any, Generator
 
 from uwtools.logging import log
-from uwtools.utils.file import readable, writable
+from uwtools.utils.file import OptionalPath, readable, writable
 
 
 def convert(
-    input_file: Optional[str] = None, output_file: Optional[str] = None, dry_run: bool = False
+    input_file: OptionalPath = None, output_file: OptionalPath = None, dry_run: bool = False
```

Comment:
I found a couple more `str` arguments to update to `OptionalPath` or `DefinitePath` to support either `str` or `Path` representations of filesystem paths.